runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerCommand.java

```diff
@@ -134,10 +148,41 @@ public void killContainer(String containerId)
     runShortCommand(Arrays.asList(dockerExecutable, "kill", containerId));
   }
 
-  /** Run the given command invocation and return stdout as a String. */
+  /**
+   * Removes docker container with container id.
+   *
+   * @throws IOException if an IOException occurs, or if the given container id either does not
+   *     exist or is still running
+   */
+  public void removeContainer(String containerId)
+      throws IOException, TimeoutException, InterruptedException {
+    checkArgument(containerId != null);
+    checkArgument(
+        CONTAINER_ID_PATTERN.matcher(containerId).matches(),
+        "Container ID must be a 64-character hexadecimal string");
+    runShortCommand(Arrays.asList(dockerExecutable, "rm", containerId));
+  }
+
   private String runShortCommand(List<String> invocation)
       throws IOException, TimeoutException, InterruptedException {
+    return runShortCommand(invocation, false, "");
+  }
+
+  /**
+   * Runs a command, blocks until {@link DockerCommand#commandTimeout} has elapsed, then returns the
+   * command's output.
+   *
+   * @param invocation command and arguments to be run
+   * @param redirectErrorStream if true, redirect stderr of the process to its stdout
+   * @param delimiter used for separating output lines
+   * @return stdout of the command, including stderr if {@code redirectErrorStream} is true
+   * @throws TimeoutException if command has not finished by {@link DockerCommand#commandTimeout}
+   */
+  private String runShortCommand(
+      List<String> invocation, boolean redirectErrorStream, CharSequence delimiter)
+      throws IOException, TimeoutException, InterruptedException {
     ProcessBuilder pb = new ProcessBuilder(invocation);
+    pb.redirectErrorStream(redirectErrorStream);
```

Comment:
We see stdout but not stderr by default? Just trying to understand why it helps to redirect stderr to stdout.

Reply:
By default, `runShortCommand` returns the stdout of the command run. Then it printed stderr iff there was an error. In my testing with the Python SDK harness, `docker logs` printed all the output we wanted to stderr, so I added the option to redirect stderr to stdout so we could capture and return it while maintaining the previous behavior for other commands.

Reply:
I suppose my question is, how will we capture the output of the `runShortCommand` when `redirectErrorStream` is set to `true`? If this simply redirects stderr to stdout, why couldn't we capture stderr before?

Reply:
Got it. Seems fair.

Reply:
Had to go and look at the entire function to understand your changes..